src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,65 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { FormEvent, useState } from 'react';
+import { createNewComment } from '../api';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  postId: number,
+  onAddComment: (newComment: Comment) => void,
+};
+
+export const NewCommentForm: React.FC<Props> = ({ postId, onAddComment }) => {
+  const [commentAuthor, setCommentAuthor] = useState('');
+  const [authorsEmail, setAuthorsEmail] = useState('');
+  const [commentText, setCommentText] = useState('');
+  const [noAuthor, setNoAuthor] = useState(false);
+  const [noEmail, setNoEmail] = useState(false);
+  const [noText, setNoText] = useState(false);
+  const [commentIsAdding, setCommentIsAdding] = useState(false);
+
+  const formSubmitHandler = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (!commentAuthor) {
+      setNoAuthor(true);
+    }
+
+    if (!authorsEmail) {
+      setNoEmail(true);
+    }
+
+    if (!commentText) {
+      setNoText(true);
+
+      return;
+    }
+
+    const newComment = {
+      postId,
+      name: commentAuthor,
+      email: authorsEmail,
+      body: commentText,
+    };
+
+    setCommentIsAdding(true);
+    createNewComment(newComment)
+      .then(res => onAddComment(res))
+      .catch()
```

Comment:
The catch block is empty. It's important to handle errors to inform the user about any issues.